#!/bin/sh -e
## 10_hppa_unaligned_constant.dpatch
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Description: Fix unaligned constant in HPPA MPI assembly (#228456)
## DP: Author: LaMont Jones <lamont@debian.org>
## DP: Upstream status: Committed for 1.2.5
## DP: Date: 2004-01-20

if [ $# -lt 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

@DPATCH@
diff -urNad /home/james/debian/packages/gnupg/gnupg-1.2.4/mpi/hppa1.1/udiv-qrnnd.S gnupg-1.2.4/mpi/hppa1.1/udiv-qrnnd.S
--- /home/james/debian/packages/gnupg/gnupg-1.2.4/mpi/hppa1.1/udiv-qrnnd.S	2002-06-29 15:14:05.000000000 +0100
+++ gnupg-1.2.4/mpi/hppa1.1/udiv-qrnnd.S	2004-02-20 16:03:54.000000000 +0000
@@ -39,6 +39,7 @@
  */
 
 	.code
+       .align          8
 	.label L$0000
 	.word		0x43f00000
 	.word		0x0
